azure_jumpstart_arcbox/artifacts/Deploy-APIM.ps1

```diff
@@ -0,0 +1,311 @@
+$WarningPreference = "SilentlyContinue"
+$ErrorActionPreference = "Stop" 
+$ProgressPreference = 'SilentlyContinue'
+# Set paths
+$Env:ArcBoxDir = "C:\ArcBox"
+$Env:ArcBoxLogsDir = "$Env:ArcBoxDir\Logs"
+$spnClientId = $env:spnClientId
+$spnClientSecret = $env:spnClientSecret
+$spnTenantId = $env:spnTenantId
+$subscriptionId = $env:subscriptionId
+$azureLocation = $env:azureLocation
+$resourceGroup = $env:resourceGroup
+
+if ($host.Name -match 'ISE') {throw "Running this script in PowerShell ISE is not supported"}
+
+try {
+    Start-Transcript -Path $Env:ArcBoxLogsDir\Deploy-APIM.log
+}
+catch {
+    Start-Transcript -Path $Env:HCIBoxLogsDir\Deploy-APIM.log
+}
+
+# Required for CLI commands
+Write-Header "Az CLI Login"
+az login --service-principal --username $Env:spnClientID --password $Env:spnClientSecret --tenant $Env:spnTenantId
+az config set extension.use_dynamic_install=yes_without_prompt
+
+################################################
+# Retrive SQL Managed Instances
+################################################
+kubectx arcbox-capi
+kubectl get nodes
+
+# Retrieving SQL MI connection endpoints
+Write-Host "`n"
+Write-Host " Retrieving SQL MI connection endpoints"
+Write-Host "`n"
+$primaryEndpoint = kubectl get sqlmanagedinstances jumpstart-sql -n arc -o=jsonpath='{.status.endpoints.primary}'
+$primaryEndpoint = $primaryEndpoint.Substring(0, $primaryEndpoint.IndexOf(',')) + ",11433" 
+$sqlConnectionString = "Data Source=$($primaryEndpoint);Initial Catalog=AdventureWorks2019;User ID=$($Env:AZDATA_USERNAME);Password=$($Env:AZDATA_PASSWORD);Encrypt=False;TrustServerCertificate=True"
+$base64ConnectionString = [Convert]::ToBase64String([char[]]$sqlConnectionString)
+
+################################################
+# Deploy AdventureWork API
+################################################
+#Switch kubectl context
+Write-Host "`n"
+Write-Host "Switch kubectl context to k3s"
+Write-Host "`n"
+
+kubectx arcbox-k3s
+kubectl get nodes
+
+# Build the adventurework api manifest
+Write-Host "`n"
+Write-Host " Build the adventurework api manifest"
+Write-Host "`n"
+
+kubectl delete secret adventurework-secrets
+$secretManifest = @"
+apiVersion: v1
+kind: Secret
+metadata:
+  name: adventurework-secrets
+type: Opaque
+data:
+  AdventureWorkConnection: $($base64ConnectionString)
+"@
+$secretManifest | kubectl apply -f -
+
+
+#Deploy AdvanetureWork API
+Write-Host "`n"
+Write-Host "Deploy AdvanetureWork API"
+Write-Host "`n"
+
+$adventureWorkManifest = @"
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: adventurework-deployment
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: adventurework
+  template:
+    metadata:
+      labels:
+        app: adventurework
+    spec:
+      containers:
+      - name: adventurework
+        image: duongthaiha/adventureworkwebapi:1.0.4
```

Comment:
Do we have this image somewhere in the MS repository? Using a personal library may break in the future.

Reply:
This is a customs image I have created for this workload. If there is a way to uploaded to a shared repo please lets me know. Happy to push it up anywhere the team see fit.

Reply:
I have update the ACR to use jumpstartprod.azurecr.io please